propresenter_bible/ui/prompting.py

```diff
@@ -0,0 +1,39 @@
+"""UI prompting helpers for interactive CLI flows."""
+
+from typing import Dict
+from prompt_toolkit import prompt
+from prompt_toolkit.completion import Completer, Completion
+from ..services.api import BibleApiClient, LanguagesResponse
+
+
+class PromptCompleter(Completer):
+    """Simple autocomplete for prompt_toolkit backed by a string map."""
+
+    def __init__(self, options: Dict[str, str]):
+        self._options = options
+
+    def get_completions(self, document, complete_event):
+        word_before_cursor = document.get_word_before_cursor()
+        for display_text, value in self._options.items():
+            if word_before_cursor.lower() in display_text.lower():
+                yield Completion(display_text, start_position=-len(word_before_cursor), display_meta=value)
+
+
+def choose_language(api: BibleApiClient) -> str:
+    """Prompt interactive language selection using the typed API client."""
+    languages: LanguagesResponse = api.get_languages_config()
+    lang_versions = languages.default_versions
+
+    def gen_name(lang_version):
+        local_name = lang_version["local_name"]
+        name = lang_version["name"]
```

Comment:
I was running into an error with these being accessed as a Dict. The below worked
`TypeError: 'LanguagesItem' object is not subscriptable`

```suggestion
        local_name = lang_version.local_name
        name = lang_version.name
```

Reply:
I don't know enough about Python to actually understand this, but it looks like these are accessed differently in the `gui` than the `import-bible` command?

```python
local_name = lang_version.get("local_name") if isinstance(lang_version, dict) else getattr(lang_version, "local_name",  "")
```